src/pay.rs

```diff
@@ -181,16 +182,25 @@ pub trait InventoryWallet: Inventory {
             }
         }
         // Add change
-        if sum_inputs > invoice.value {
-            let seal = output_for_assignment(suppl.as_ref(), assignment_id)?;
-            let change = TypedState::Amount(sum_inputs - invoice.value);
-            main_builder = main_builder.add_raw_state(assignment_id, seal, change)?;
-        } else if sum_inputs < invoice.value {
-            return Err(PayError::InsufficientState);
-        }
-        let transition = main_builder
-            .add_raw_state(assignment_id, beneficiary, TypedState::Amount(invoice.value))?
-            .complete_transition()?;
+        let transition = match invoice.value {
+            TypedState::Amount(amt) => {
+                match sum_inputs.cmp(&amt) {
+                    Ordering::Greater => {
+                        let seal = output_for_assignment(suppl.as_ref(), assignment_id)?;
+                        let change = TypedState::Amount(sum_inputs - amt);
+                        main_builder = main_builder.add_raw_state(assignment_id, seal, change)?;
+                    }
+                    Ordering::Less => return Err(PayError::InsufficientState),
+                    Ordering::Equal => {}
+                }
+                main_builder
+                    .add_raw_state(assignment_id, beneficiary, TypedState::Amount(amt))?
+                    .complete_transition()?
+            }
+            _ => {
+                unimplemented!("only TypedState::Amount is currently supported")
```

Comment:
```suggestion
                todo!("only TypedState::Amount is currently supported")
```

`unimplemented` in rust means "never going to be implemented". It is also not listed as TODOs by many IDEs.